packages/cli/src/commands/replay/utils/compute-diff.ts

```diff
@@ -27,42 +33,128 @@ export interface ComputeAndSaveDiffOptions {
-    getReplayDir(baseReplayId)
+  const screenshotIdentifiersByBaseReplayId =
+    groupByBaseReplayId(baseScreenshots);
+  await downloadScreenshots(baseScreenshots, baseScreenshotsDir);
+  logger.debug(
+    `Downloaded ${baseScreenshots.length} base screenshots to ${baseScreenshotsDir}}`
   );
+
   const headReplayScreenshotsDir = getScreenshotsDir(tempDir);
+  const headReplayScreenshots = await getScreenshotFiles(
+    headReplayScreenshotsDir
+  );
+  const screenshotDiffResultsByBaseReplayId: Record<
+    string,
+    ScreenshotDiffResult[]
+  > = {};
 
-  const screenshotDiffResults = await diffScreenshots({
-    client,
-    baseReplayId,
-    headReplayId,
-    baseScreenshotsDir: baseReplayScreenshotsDir,
-    headScreenshotsDir: headReplayScreenshotsDir,
+  // Diff the screenshots for each base replay
+  for (const baseReplayId of screenshotIdentifiersByBaseReplayId.keys()) {
+    const baseScreenshotIdentifiers =
+      screenshotIdentifiersByBaseReplayId.get(baseReplayId) ?? [];
+    const baseReplayScreenshots = await getFilteredScreenshotFiles(
+      baseScreenshotsDir,
+      baseScreenshotIdentifiers
+    );
+    const resultsForBaseReplay = await diffScreenshots({
+      client,
+      baseReplayId,
+      headReplayId,
+      baseScreenshotsDir: baseScreenshotsDir,
+      headScreenshotsDir: headReplayScreenshotsDir,
+      headReplayScreenshots,
+      baseReplayScreenshots,
+      diffOptions: screenshottingOptions.diffOptions,
+      logger,
+    });
+    screenshotDiffResultsByBaseReplayId[baseReplayId] = resultsForBaseReplay;
+  }
+
+  logDifferences({
+    results: Object.values(screenshotDiffResultsByBaseReplayId).flat(),
     diffOptions: screenshottingOptions.diffOptions,
     logger,
   });
 
-  const screenshotDiffsSummary = summarizeDifferences({
-    baseReplayId,
-    headReplayId,
-    results: screenshotDiffResults,
-    diffOptions: screenshottingOptions.diffOptions,
-    logger,
+  return screenshotDiffResultsByBaseReplayId;
+};
+
+const groupByBaseReplayId = (screenshots: ScreenshotLocator[]) => {
+  const screenshotIdentifiersByBaseReplayId = new Map<
+    string,
+    ScreenshotIdentifier[]
+  >();
+  screenshots.forEach(({ replayId, screenshotIdentifier }) => {
+    const screenshotIdentifiers =
+      screenshotIdentifiersByBaseReplayId.get(replayId) ?? [];
+    screenshotIdentifiers.push(screenshotIdentifier);
+    screenshotIdentifiersByBaseReplayId.set(replayId, screenshotIdentifiers);
   });
+  return screenshotIdentifiersByBaseReplayId;
+};
+
+const downloadScreenshots = async (
+  screenshots: ScreenshotLocator[],
+  directory: string
+) => {
+  // Download in batches of 20
+  const screenshotBatches = chunk(screenshots, 20);
+  for (const screenshotBatch of screenshotBatches) {
+    const screenshotDownloads = screenshotBatch.map((screenshot) => {
+      const filePath = join(
+        directory,
+        basename(new URL(screenshot.screenshotUrl).pathname)
```

Comment:
Should we use the screenshot identifier here to generate a file name? I don't think we should assume that / rely on screenshot URL paths matching the identifier format.